Per-user track changes

paulojreis opened the pull request:
We'll need to change https://github.com/sharelatex/web-sharelatex/pull/547/files#diff-7701684cedd84223921f73f784273c4eR239 to the date of the deploy just before going live

jpallen commented:
At first, when turning on 'everyone', the toggles are selected. On refresh they are grey:

<img width="238" alt="screen shot 2017-07-21 at 13 45 49" src="https://user-images.githubusercontent.com/31305/28463971-14484e7e-6e1b-11e7-9301-f48d9eba361c.png">

jpallen commented on app/coffee/Features/Project/ProjectController.coffee:
```diff
@@ -230,14 +230,32 @@ module.exports = ProjectController =
+					return cb()
+				timestamp = user_id.toString().substring(0,8)
+				userSignupDate = new Date( parseInt( timestamp, 16 ) * 1000 )
+				if userSignupDate > new Date("2018-01-01") # 8th March
```
Needs updating

jpallen commented on public/coffee/ide/review-panel/controllers/ReviewPanelController.coffee:
```diff
@@ -565,24 +583,86 @@ define [
-				$scope.editor.wantTrackChanges = value
+				applyTrackChangesStateToClient(state)
+
+		# Not sure what the kbd shortcut should do now?
```
Did we have any solutions for this? Toggle it for the user maybe?

paulojreis commented:
@jpallen , just addressed all your points:

* Toggles are "on" or "off" for all users, when the setting is applied to everyone, even on project load/refresh;
* Keyboard shortcut is re-enabled, toggles TC for the user;
* Sign-up date threshold to show the per-user TC prompt is updated.

Can you take a look again?
